lib/internal/per_context/primordials.js

```diff
@@ -120,13 +119,7 @@ primordials.SafePromise = makeSafe(
   'WeakSet',
 ].forEach((name) => {
   const original = global[name];
-  primordials[name] = Object.setPrototypeOf({
-    [name]: function(...args) {
-      return new.target ?
-        ReflectConstruct(original, args, new.target) :
-        ReflectApply(original, this, args);
-    }
-  }[name], null);
```

Comment:
This change is necessary, for two reasons:
- Some of the arrays we create are returned to the user
- Without it I'd also have to change calls to array methods
Also, It's probably better not to wrap the primordial functions for performance reasons, especially when we migrate String() and Number() calls in the future.

Reply:
hmm, wait, but why is `Object.setPrototypeOf(..., null)` removed here?

Reply:
Because if we stop wrapping the functions, it becomes useless, as there is nothing left to set the prototype of (we can't do that directly on the original functions).

Reply:
I see. To me making the Array namespace less mutable takes precedence over keeping these functions away from prototype poisoning, so LGTM.